src/main/java/digital/slovensko/autogram/util/Version.java

```diff
@@ -0,0 +1,76 @@
+package digital.slovensko.autogram.util;
+
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+public class Version implements Comparable<Version> {
+
+    private int[] versionNumbers;
+    private boolean dev;
+
+    public Version(int[] versionNumbers) {
+        this.versionNumbers = versionNumbers;
+    }
+
+    public static Version createFromVersionString(String version) {
+        version = version.replaceAll("^v", "");
+
+        if (version.equals("")) {
+            return new Version(new int[]{});
+        }
+
+        if (version.equals("dev")) {
+            return createDevVersion();
+        }
+
+        var versionNumbers = Stream.of(version.split("\\.")).mapToInt(Integer::parseInt).toArray();
+        return new Version(versionNumbers);
+    }
+
+    private static Version createDevVersion() {
+        var v = new Version(new int[]{});
+        v.dev = true;
```

Comment:
Spravíme konštruktor bez parametrov a vtedy to bude dev a môžeme mať všetky fields final?

Reply:
dobry napad